lua_modules/ftp/server.lua

```diff
@@ -0,0 +1,157 @@
+-- a simple ftp server
+USER = "test"
+PASS = "12345"
+local file,net,pairs,print,string,table = file,net,pairs,print,string,table
+data_fnc = nil
+ftp_data = net.createServer(net.TCP, 180)
+ftp_data:listen(20, function (s)
+  if data_fnc then data_fnc(s) end
+end)
+ftp_srv = net.createServer(net.TCP, 180)
+ftp_srv:listen(21, function(socket)
+  local s = 0
+  local cwd = "/"
+  local buf = ""
+  local t = 0
+  socket:on("receive", function(c, d)
+    a = {}
+    for i in string.gmatch(d, "([^ \r\n]+)") do
+      table.insert(a,i)
+    end
+    if(a[1] == nil or a[1] == "")then return end
+    if(s == 0 and a[1] == "USER")then
```

Comment:
You know that the Lua syntax is `if cond then ...`.  This isn't C or Javascript :smiley:. IMO  `if a2 ~= USER then`is clearer, but even if you want to keep the () or even ((())) then add the spaces to make it readable.

Reply:
Yes - wery more JS and Cpp make me use ```If()``` automaticly